src/components/dateTimePicker/index.tsx

```diff
@@ -199,72 +186,58 @@ class DateTimePicker extends Component<DateTimePickerPropsInternal, DateTimePick
     }
   };
 
-  getDialogProps = () => {
-    const {testID, dialogProps} = this.props;
-    return {
-      width: '100%',
-      height: null,
-      bottom: true,
-      centerH: true,
-      // onDismiss: this.toggleExpandableOverlay,
-      containerStyle: styles.dialog,
-      testID: `${testID}.dialog`,
-      supportedOrientations: [
-        'portrait',
-        'landscape',
-        'landscape-left',
-        'landscape-right'
-      ] as DialogProps['supportedOrientations'],
-      ...dialogProps
-    };
-  };
+  const toggleExpandableOverlay = useCallback(() => {
+    expandable.current?.toggleExpandable?.();
+  }, []);
 
-  renderIOSExpandableOverlay = () => {
-    return (
-      <>
-        {this.renderHeader()}
-        {this.renderDateTimePicker()}
-      </>
-    );
-  };
+  const onDonePressed = useCallback(() => {
+    toggleExpandableOverlay();
+    if (Constants.isIOS && !chosenDate.current) {
+      // since handleChange() is not called on iOS when there is no actual change
+      chosenDate.current = new Date();
+    }
 
-  renderHeader() {
-    // @ts-expect-error
-    const {headerStyle, useCustomTheme} = this.props;
+    onChange?.(chosenDate.current!);
+    setValue(chosenDate.current);
+  }, [toggleExpandableOverlay, onChange]);
```

Comment:
Don't you need to add `chosenDate` to the dependency array?

Reply:
`chosenDate` is a ref, it has no effect on the callback.